Patch from 
   https://github.com/JuliaLang/julia/commit/a2d85f6d.patch


From a2d85f6dc63439aea9f973bacb55006596fdb2b8 Mon Sep 17 00:00:00 2001
From: Fangrui Song <i@maskray.me>
Date: Tue, 4 May 2021 11:53:40 -0700
Subject: [PATCH] Replace llvm::sys::fs::F_None with llvm::sys::fs::OF_None
 (#40666)

The former is deprecated.
OF_None has been available in LLVM since 2018-06.
---
 src/codegen.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/codegen.cpp b/src/codegen.cpp
index ee5ec4dce0c1..bc60798220d5 100644
--- a/src/codegen.cpp
+++ b/src/codegen.cpp
@@ -8092,12 +8092,12 @@ extern "C" void jl_dump_llvm_mfunction(void *v)
 
 extern void jl_write_bitcode_func(void *F, char *fname) {
     std::error_code EC;
-    raw_fd_ostream OS(fname, EC, sys::fs::F_None);
+    raw_fd_ostream OS(fname, EC, sys::fs::OF_None);
     llvm::WriteBitcodeToFile(*((llvm::Function*)F)->getParent(), OS);
 }
 
 extern void jl_write_bitcode_module(void *M, char *fname) {
     std::error_code EC;
-    raw_fd_ostream OS(fname, EC, sys::fs::F_None);
+    raw_fd_ostream OS(fname, EC, sys::fs::OF_None);
     llvm::WriteBitcodeToFile(*(llvm::Module*)M, OS);
 }
